_drafts/2020-08-07-g2p.md

```diff
@@ -0,0 +1,625 @@
+
+You can recreate this type of animation using the G2P studio by selecting the French (`fra`) to English Arpabet (`eng-arpabet`) mapping, choosing 'animate' and then typing your text in the input area as seen below:
+
+{% picture g2p-animate.png %}
+
+#### What can you do with your aligned audio/text?
+
+ReadAlongs exports to a variety of formats, including epub (for e-readers), [Praat](https://www.fon.hum.uva.nl/praat/) TextGrids, [ELAN](https://archive.mpi.nl/tla/elan) files, various subtitle formats, and an embeddable web component for your website. This project is still *very* unstable though, and it is **not** yet recommended to use it in production.
+
+Below is an example of the embeddable web component in Danish.
+
+{% include readalong.html text="/assets/ras/output.xml" alignment="/assets/ras/output.smil" audio="/assets/ras/output.wav" %}
+
+### Convertextract
+
+Convertextract is a tool that lets you use `g2p` mappings to convert text inside a variety of different documents. Have a look at the [convertextract repo readme](https://github.com/roedoejet/convertextract) or at Fineen Davis' [blog post]({{ "convertextract-app" | absolute_url }}) about the Convertextract GUI.
+
+### Advanced: use in Python and the command line
+
+#### Installation
+
+We recommend installing `g2p` using [pip](https://en.wikipedia.org/wiki/Pip_(package_manager)):
+
+```python
+pip install g2p
+```
+
+If you are going to be creating your own mapping or editing `g2p` in any way, you must first fork `g2p` by going to [https://github.com/roedoejet/g2p](https://github.com/roedoejet/g2p) and forking the project to your own GitHub space.
+
+Once you've forked `g2p`, clone your own fork:
+
+```bash
+git clone https://github.com/YourGitHubUsername/g2p.git
+```
+
+and then do an *editable* pip installation:
+
+```bash
+cd g2p && pip install -e .
+```
+
+#### Usage
+
+Using `g2p` within Python can be done, either programatically using the `make_g2p` function:
+
+```python
+>>> from g2p import make_g2p
+>>> transducer = make_g2p('dan', 'eng-arpabet')
+>>> transducer('hej').output_string
+'HH EH Y'
+```
+
+#### Command Line Interface
+
+You can also use `g2p` from the command line. The basic command for conversions is:
+
+```bash
+g2p convert <input_text> <in_lang> <out_lang>
+```
+
+So in practice:
+
+```bash
+$ g2p convert hej dan eng-arpabet
+HH EH YY
+```
+
+## Advanced: contributing your rules to the main `g2p` library
+
+You've written some cool rules and you want to contribute, that's awesome! There are lots of benefits to contributing your mapping to `g2p`. First of all, once your mapping is accepted, you'll have it available and live on [G2P Studio](https://g2p-studio.herokuapp.com). Second, once the next version of `g2p` is released with your mapping, it will be automatically built in to the [Convertextract](#convertextract) library. Third, if your mapping is between a language's writing system and the IPA, you can also get ReadAlongs support for your language.
```

Comment:
```suggestion
You've written some cool rules and you want to contribute, that's awesome! There are lots of benefits to contributing your mapping to `g2p`. First of all, submit your mapping to have it accepted (for instructions on how, see below). Once your mapping is accepted, you'll have it available and live on [G2P Studio](https://g2p-studio.herokuapp.com). Second, once the next version of `g2p` is released with your mapping, it will be automatically built in to the [Convertextract](#convertextract) library. Third, if your mapping is between a language's writing system and the IPA, you can also get ReadAlongs support for your language.
```